### PR TITLE
Support `.removeDuplicates()` for `Defaults.publisher`

### PR DESCRIPTION
**Pull request by IuriiIaremenko**
Support `removeDuplicates()` combine operator for  Defaults  publishers #43 

```swift
	Defaults
	   .publisher(key, options: [])
	   .removeDuplicates()
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#43: Add `removeDuplicates` option to `Defaults.publisher(keys...)` and `Defaults.publisherAll()`](https://issuehunt.io/repos/129132562/issues/43)
---
</details>

### CODE REVIEW

**Comment by IuriiIaremenko** [comment]
@sindresorhus please review. Failing test case is not related to the change
 ```
/Users/runner/work/Defaults/Defaults/Tests/DefaultsTests/DefaultsTests.swift:607: error: -[DefaultsTests.DefaultsTests testObservePreventPropagationMultipleThreads] : Asynchronous wait failed: Exceeded timeout of 10 seconds, with unfulfilled expectations: "No infinite recursion".
```

**Comment by IuriiIaremenko** [comment]
@sindresorhus do you have a spare minute to review the PR?

File: Sources/Defaults/Observation+Combine.swift [hunk 1]
```diff
@@ -151,4 +151,24 @@ extension Defaults {
 		return combinedPublisher
 	}
 }
+
+@available(macOS 10.13, OSXApplicationExtension 10.13, watchOSApplicationExtension 4.0, *)
+extension Defaults.NSSecureCodingKeyChange: Equatable where Value: Equatable {
+	public static func == (lhs: Self, rhs: Self) -> Bool {
+		lhs.newValue == rhs.newValue
```
Comment (sindresorhus) [inline Sources/Defaults/Observation+Combine.swift:158]: While this works for your use-case, I don't think this is correct. It should be equatable on all the properties. You can just rely on compiler synthesis for that.

Comment (IuriiIaremenko) [reply Sources/Defaults/Observation+Combine.swift]: @sindresorhus got it, fixed.

**Pull request merged**